Add "crms2map"

jinikeda opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jinikeda commented:
Ready to review @conda-forge/help-python I appreciate your support.

synapticarbors commented on recipes/crms2map/LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
The license file is already in the source distribution so you don't need to include it again here.

jinikeda replied:
Delete the file.

synapticarbors commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,108 @@
+  sha256: 7c718dbe50dd3fe7cb2cf294418e846028126ed6a4f9c3960db92b7cfce3ae53
+
+build:
+  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
```
Please remove all comments that aren't specific to this recipe

jinikeda replied:
Removed all comments that are not related to the recipe.

synapticarbors commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,108 @@
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  # More info about selectors can be found in the conda-build docs: 
+  # https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors
+  noarch: generic
```
This should probably be `noarch: python`

jinikeda replied:
Modified. Thank you.

synapticarbors commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,108 @@
+  url: https://github.com/jinikeda/{{ name |lower }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 7c718dbe50dd3fe7cb2cf294418e846028126ed6a4f9c3960db92b7cfce3ae53
+
+build:
```
This project appears to have a cli entry point. Please include it and test it:
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

jinikeda replied:
Added entry point and test command. Thank you.

synapticarbors commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,108 @@
+    - python >=3.9
+    - pip
+    - setuptools
+  run:
```
You appear to be adding many requirements that aren't listed upstream:
https://github.com/jinikeda/crms2map/blob/main/setup.py#L14-L24

jinikeda replied:
I modified the "run:" script and setup.py in my repository (which now coincides with them). Please let me know if I need to make the source code again. Thanks.

synapticarbors commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,108 @@
+  requires:
+    - pytest
+    - pytest-mock
+    - click
```
You don't need to list requirements here that are already in the main `run` requirements. Those get pulled in automatically.

jinikeda replied:
I adjusted the test requirements.

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

xhochy committed:
Merge branch 'main' into add-crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

jinikeda commented:
Although it merged branch 'main' into add-crms2map on Sep 22, 2024, I am not sure if I still need to the change request or not. please let me know.

xhochy commented:
You should address the lint.

jikeda committed:
Add/Modify recipe for crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/crms2map/meta.yaml```) and found it was in an excellent condition.

jinikeda commented:
Modified the linter comments.

jinikeda committed:
Merge branch 'main' into add-crms2map

xhochy commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,70 @@
+{% set name = "crms2map" %}
+{% set version = "1.0" %}
+{% set python_min = "3.9" %}
```
This is the default and thus should be omitted.
```suggestion
```

jinikeda replied:
Modified. Thank you.

xhochy commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,70 @@
+  number: 0
+
+requirements:
+  build:
```
```suggestion
```

jinikeda replied:
Modified.

xhochy commented on recipes/crms2map/meta.yaml:
```diff
@@ -0,0 +1,70 @@
+
+test:
+  imports:
+    - src.CRMS_general_functions
```
This doesn't look right, packages should not claim the `src` folder.

jinikeda replied:
Thank you for your feedback regarding the src folder in the meta.yaml file. I slightly modified meta.yaml. The test codes are designed for simple functional tests for individual components rather than a comprehensive package test.

xhochy replied:
Please address this. This is a issue with the upstream package but as you are upstream, you should fix that yourself.

jinikeda replied:
Thank you for your comment. Based on your comment, I fixed the code. Thank you.

jikeda committed:
Add recipe for crms2map

jikeda committed:
Merge branch 'add-crms2map' of github.com:jinikeda/staged-recipes into add-crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Add recipe for crms2map

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/crms2map/meta.yaml```) and found it was in an excellent condition.

jinikeda committed:
Merge branch 'main' into add-crms2map

jikeda committed:
Add recipe for crms2map

jikeda committed:
Merge branch 'add-crms2map' of github.com:jinikeda/staged-recipes into add-crms2map

jikeda committed:
Add recipe for crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

xhochy commented:
You are continuously ignoring my comments. I will no longer comment on this PR.

jinikeda commented:
> You are continuously ignoring my comments. I will no longer comment on this PR.
Dear xhochy
 Thank you for your review. I am confused about receiving this message because I fixed it and replied to your message on Dec 28.
 My modification might still be something wrong, and I also don't fully understand the receipt. However, I have tried to respond as honestly as possible. Would you kindly review my latest update on Dec 28th?
I really appreciate your kindness.

[jinikeda](https://github.com/jinikeda) [on Dec 28, 2024](https://github.com/conda-forge/staged-recipes/pull/27371#discussion_r1898982395)
Thank you for your comment. Based on your comment, I fixed the code. Thank you.

jinikeda committed:
Merge branch 'main' into add-crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

jinikeda committed:
Merge branch 'main' into add-crms2map

matthewfeickert commented:
> Can someone please review my pull request?

@jinikeda Please address the issues that were raised such as https://github.com/conda-forge/staged-recipes/pull/27371/changes#r1878745597. The current imports in this PR are still problematic.
